pkg/gardener/shoot/extender/provider.go

```diff
@@ -98,8 +98,10 @@ func NewProviderExtenderPatchOperation(enableIMDSv2 bool, defMachineImgName, def
 			return err
 		}
 
+		correctOneNodeWorkerPools(provider, shootWorkers)
```

Comment:
(didn't know a better place to add this question)
Should we also consider reverting #682 ?